Shared/KeychainCache.swift

```diff
@@ -30,13 +30,13 @@ open class KeychainCache<T: JSONLiteralConvertible> {
     open class func fromBranch(_ branch: String,
                                withLabel label: String?,
                                withDefault defaultValue: T? = nil,
-                               factory: (JSON) -> T?,
+                               factory: ([String: Any]) -> T?,
                                logger: Logger = DefaultLogger.shared) -> KeychainCache<T> {
         if let l = label {
             let key = "\(branch).\(l)"
             MZKeychainWrapper.sharedClientAppContainerKeychain.ensureStringItemAccessibility(.afterFirstUnlock, forKey: key)
             if let s = MZKeychainWrapper.sharedClientAppContainerKeychain.string(forKey: key) {
-                if let t = factory(JSON(parseJSON: s)) {
+                if let dictionaryObject = JSON(parseJSON: s).dictionaryObject, let t = factory(dictionaryObject) {
```

Comment:
I almost commented on this, only to realize `JSON(parseJSON` should be done as part of https://github.com/mozilla-mobile/firefox-ios/issues/13376 haha. So commenting in case someone else sees this and wonder